src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1345,6 +1445,11 @@ private void AssignPatternVariables(BoundPattern pattern)
         public override BoundNode VisitBlock(BoundBlock node)
         {
             DeclareVariables(node.Locals);
+
+            // First assemble possibly-unassigned reads for all
+            // local functions
+            RecordReadsInLocalFunctions(node);
```

Comment:
We prefer to avoid quadratic or exponential algorithms.

I have to say I'm not sure I fully understand the intended phase ordering.

---
In reply to: [74492362](https://github.com/dotnet/roslyn/pull/12888#discussion_r74492362) [](ancestors = 74492362)